src/v0/destinations/criteo_audience/util.js

```diff
@@ -0,0 +1,76 @@
+const { isDefinedAndNotNullAndNotEmpty } = require("../../util");
+const {
+  ConfigurationError,
+  OAuthSecretError,
+  InstrumentationError
+} = require("../../util/errorTypes");
+/**
+ * Get access token to be bound to the event req headers
+ *
+ * Note:
+ * This method needs to be implemented particular to the destination
+ * As the schema that we'd get in `metadata.secret` can be different
+ * for different destinations
+ *
+ * @param {Object} metadata
+ * @returns
+ */
+const getAccessToken = metadata => {
```

Comment:
We can move this to a utility and modify the signature in this way
```javascript
// v0/util/index.js
const getAccessToken = (metadata, accessTokenKey) => {
  // OAuth for this destination
  const { secret } = metadata;
  // we would need to verify if secret is present and also if the access token field is present in secret
  if (!secret || !secret[accessTokenKey]) {
    throw new OAuthSecretError("Empty/Invalid access token");
  }
  return secret[accessTokenKey];
};

// criteo_audience/transform.js
const processEvent = async (metadata, message, destination) => {
  const accessToken = getAccessToken(metadata, "accessToken");
  ...
}
```
We can extend this method to other OAuth destinations as part of another PR

Reply:
We can do it but some dest may take it in another form. like mentioning it as accessTokenKey or some simply accessToken. To tackle this we have to change the rudderauth return param name. I guess this can be taken care in a different PR

Reply:
No, we shouldn't put anything in rudder-auth or some other service. As this is destination specific key that they send as part of refresh or initial token generation step, we shouldn't keep it as a convention.

Reply:
Yeah so will need to think of another way

Reply:
Implemented a common utility method to get accessToken for OAuth destinations